<?xml version="1.0"?>


<launch>
  <arg name="base_frame" default="base_footprint"/>
  <arg name="fixed_frame" default="map"/>
  <arg name="bot_params" default="turtlebot_params" />  
  <arg name="map" default="maze.yaml" />
  <arg name="joy" default="remote"/>   <!--"user" or "remote"-->

    <!-- Launch hardware drivers, alternatively, use the 'bare' simulation launch files  in mbot_navigation or scatwheel_navigation -->
    <!-- <include file="$(find scat_launch)/launch/system.launch">
    </include> -->

    <!-- Launch Localization pipeline -->
    <include file="$(find scat_launch)/launch/localization.launch">
	  <arg name="map" value="$(arg map)" />
    </include>

    <!-- Velocity Command filter -->
  <node pkg="vel_safety_filter" type="vel_safety_filter" name="vel_safety_filter" output="screen">
    <rosparam command="load" file="$(find vel_safety_filter)/config/settings.yaml" />    
    <rosparam command="load" file="$(find scat_launch)/config/$(arg bot_params).yaml" />
    <remap from="/odom" to="/ekf/odom" />
    <param name="use_vel_filter" value="false"/>
    <param name="use_col_filter" value="true"/>
  </node>

    <!-- move_base node -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
    <remap from="/odom" to="/ekf/odom"/>
    <remap from="/cmd_vel" to="/control/cmd_vel"/>

    <param name="base_global_planner" value="global_planner_plugin/GlobalPlannerPlugin" />    
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />

    <rosparam file="$(find scat_launch)/config/move_base_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/local_planner_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find scat_launch)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find scat_launch)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/global_costmap_params.yaml" command="load" />
    <param name="planner_frequency" value="0.5" />
  </node>

    <!-- Joystick controllers. Use either joystick_filter or control_joystick (no filter) -->
  <node pkg="joystick_filter" type="joystick_filter" name="joystick_filter" output="screen">
    <rosparam command="load" file="$(find joystick_filter)/config/settings.yaml" />
      <remap from="/user/joy" to ="/$(arg joy)/joy" />
  </node>

  <!-- Sparse voronoi trajectory generator --> 
  <node pkg="sparse_voronoi" type="sparse_voronoi" name="sparse_voronoi" output="screen">
    <rosparam command="load" file="$(find sparse_voronoi)/config/settings.yaml" />
    <remap from="/sparse_voronoi/interp_path" to="/global_planner_plugin/trajectory" />
    <remap from="/sparse_voronoi/goal" to="/move_base_simple/goal" />
    <param name="interface_movebase" value="true"/>
    <param name="pub_frequency" value="2"/>
    <param name="calc_frequency" value="2"/>
  </node>

  <!-- Launch Obstacle Detector --> 
  <!-- <include file="$(find obstacle_detector)/launch/nodes.launch"/> -->

  <!-- Launch Line Detector -->
  <!-- <include file="$(find laser_line_extraction)/launch/line_extractor.launch"/> -->

  <!-- Launch Line Detector --> 
  <include file="$(find laserline)/launch/laserline.launch"/> 
  
</launch>
